modelbaker/ilitoppingmaker/iliprojecttopping.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+"""
+/***************************************************************************
+                              -------------------
+        begin                : 2022-07-17
+        git sha              : :%H$
+        copyright            : (C) 2022 by Dave Signer
+        email                : david at opengis ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+
+
+from qgis.core import Qgis, QgsProject
+
+from ..libs.toppingmaker import ExportSettings, ProjectTopping
+from .ilidata import IliData
+from .ilitarget import IliTarget
+from .metaconfig import MetaConfig
+
+
+class IliProjectTopping(ProjectTopping):
+    """
+    A project configuration resulting in a YAML file that contains:
+    - layertree
+    - layerorder
+    - project variables (future)
+    - print layout (future)
+    - map themes (future)
+    QML style files, QLR layer definition files and the source of a layer can be linked in the YAML file and are exported to the specific folders.
+
+    Optimised for INTERLIS projects having artefacts like ilidata, metaconfigfile (ini) etc. with methods to generate them.
+    """
+
+    def __init__(
+        self,
+        target=IliTarget(),
+        export_settings: ExportSettings = ExportSettings(),
+        metaconfig=MetaConfig(),
+    ):
+        # while ProjectTopping does not hold objects like target and export_settings (and those are passed by the function) the IliProjectTopping keeps it to set up the whole IliProjectTopping before create anything.
+        super().__init__()
+        self.target = target
+        self.export_settings = export_settings
+        self.metaconfig = metaconfig
+
+    @property
+    def models(self):
+        return self.metaconfig.ili2db_settings.models
+
+    @property
+    def referencedata_paths(self):
+        return self.metaconfig.referencedata_paths
+
+    def set_models(self, models: list = []):
```

Comment:
```suggestion
    @models.setter
    def models(self, models: list):
```

If using properties, why not also for setting?
Is there a reason for the default value in here?

Reply:
Definitely, thanks. Yes. no need for defaults.